ComponentKit/Core/CKComponentMemoizer.h

```diff
@@ -56,15 +61,27 @@ id CKMemoize(CKMemoizationKey memoizationKey, id (^block)(void));
 
  }
  
- How to use component layout memoization, override -shouldMemoizeLayout in your component subclass.
- See CKComponentSubclass.h for more info.
+ How to use component layout memoization, override computeLayoutThatFits:restrictedToSize:relativeToParentSize: in your
+ component subclass, and call super within the block you provide to CKMemoizeLayout. The results of the computation for
+ your component at these constraints will be saved and returned without calling your provided block, if possible.
+ 
+ - (CKComponentLayout)computeLayoutThatFits:(CKSizeRange)constrainedSize 
+                           restrictedToSize:(const CKComponentSize &)size
+                       relativeToParentSize:(CGSize)parentSize
+ {
+   return CKMemoizeLayout(self, constrainedSize, size, parentSize, ^CKComponentLayout{
+     return [super computeLayoutThatFits:constrainedSize
+                        restrictedToSize:size
+                    relativeToParentSize:parentSize];
+   });
+ }
```

Comment:
It may be a good idea to add these comments to the `CKMemoizeLayout` function itself.

How much of this header is actually part of the public-facing API? I ask because there is a fair amount of stuff here and we should take this opportunity to clean this up a bit if we want to.